chalk-solve/src/clauses.rs

```diff
@@ -168,8 +179,8 @@ fn program_clauses_that_could_match<I: Interner>(
                             push_auto_trait_impls(builder, trait_id, struct_id);
                         }
                     }
-                    TyData::InferenceVar(_) => {
-                        panic!("auto-traits should flounder if nothing is known")
+                    TyData::InferenceVar(_) | TyData::BoundVar(_) => {
```

Comment:
Seems like maybe you did... that seems like a bug somewhere -- oh, right I think you said you are invoking this in the recursive solver before doing instantiation? Not sure what I think of that but maybe it's ok.